v3/src/components/graph/adornments/region-of-interest/region-of-interest-adornment-model.ts

```diff
@@ -0,0 +1,52 @@
+import { Instance, SnapshotIn, types } from "mobx-state-tree"
+import { AdornmentModel, IAdornmentModel } from "../adornment-models"
+import { kRegionOfInterestType } from "./region-of-interest-adornment-types"
+
+export type RoiPositionUnit = "coordinate" | "percent" | "%"
+
+export const RoiUnitValueModel = types.model("RoiPositionModel", {
+  unit: types.enumeration<RoiPositionUnit>("unit", ["coordinate", "percent", "%"]),
+  value: types.number
+})
+
+export const RegionOfInterestAdornmentModel = AdornmentModel
+  .named("RegionOfInterestAdornmentModel")
+  .props({
+    type: types.optional(types.literal(kRegionOfInterestType), kRegionOfInterestType),
+    height: types.maybe(RoiUnitValueModel),
+    width: types.maybe(RoiUnitValueModel),
+    xAttribute: types.optional(types.string, ""),
+    xPosition: types.maybe(RoiUnitValueModel),
+    yAttribute: types.optional(types.string, ""),
+    yPosition: types.maybe(RoiUnitValueModel)
```

Comment:
This model representation will result in the ROI adornment being less robust to graph configuration changes than other adornments. See suggestion elsewhere for how to improve it.